packages/jellyfish-testing/src/index.ts

```diff
@@ -81,3 +84,56 @@ class TestingJsonRpcClient extends JsonRpcClient {
     })
   }
 }
+
+export class TestingGroup {
+  private constructor (
+    public readonly group: ContainerGroup,
+    public readonly testings: Testing[]
+  ) {
+  }
+
+  static create (n: number, init: (index: number) => MasterNodeRegTestContainer): TestingGroup {
```

Comment:
We still need the ability to call `TestingGroup.create(3)` for default container types. i.e. `init()` should be optional. please check if there's a better syntax to write this. but this one works!
```suggestion
  static create (n: number, init: (index: number) => MasterNodeRegTestContainer = (index) => new MasterNodeRegTestContainer(GenesisKeys[index])): TestingGroup {
```

Reply:
yes will change to optional
but this one is the most simplified ady 
```init: (index: number) => MasterNodeRegTestContainer)```

Reply:
clarification needed.
then where is the default function body? what you are giving as `init: (index: number) => MasterNodeRegTestContainer)` is the type of the `init` and not the default function. am I missing something, thinking from c++ perspective 😅

Reply:
init: (index: number) => MasterNodeRegTestContainer = (index) => new MasterNodeRegTestContainer(GenesisKeys[index])
is same as 
init: (index: number) => MasterNodeRegTestContainer (a shortcut to write)

Reply:
We still need the default functions, let me create a PR into this branch

Reply:
@fuxingloh i can update the changes here.. i haven't got idea of setting the default func there yet

Reply:
FYI,  setting up default func body is not allowed to set optional param
![image](https://user-images.githubusercontent.com/8097901/132465904-8ba971da-f65f-4afd-b1af-12c188e5d849.png)

I refined the optional mnkeys logic for easier

Reply:
for this one do not need to put `?`, having the initializer means if you don't provide it outside, it will use the default.